src/OpenTelemetry/Internal/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,164 @@
+// <copyright file="ConfigurationExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+#if !NETFRAMEWORK && !NETSTANDARD2_0
+using System.Diagnostics.CodeAnalysis;
+#endif
+using System.Globalization;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace OpenTelemetry.Internal;
+
+internal static class ConfigurationExtensions
+{
+    public delegate bool TryParseFunc<T>(
+        string value,
+#if !NETFRAMEWORK && !NETSTANDARD2_0
+        [NotNullWhen(true)]
+#endif
+        out T? parsedValue);
+
+    public static bool TryGetStringValue(
+        this IConfiguration configuration,
+        string key,
+#if !NETFRAMEWORK && !NETSTANDARD2_0
+        [NotNullWhen(true)]
+#endif
+        out string? value)
+    {
+        value = configuration.GetValue<string?>(key, null);
+
+        return !string.IsNullOrWhiteSpace(value);
+    }
+
+    public static bool TryGetUriValue(
+        this IConfiguration configuration,
+        string key,
+#if !NETFRAMEWORK && !NETSTANDARD2_0
+        [NotNullWhen(true)]
+#endif
+        out Uri? value)
+    {
+        if (!configuration.TryGetStringValue(key, out var stringValue))
+        {
+            value = null;
+            return false;
+        }
+
+        if (!Uri.TryCreate(stringValue, UriKind.Absolute, out value))
+        {
+            throw new FormatException($"{key} environment variable has an invalid value: '{stringValue}'");
```

Comment:
Just a bug reminder https://github.com/open-telemetry/opentelemetry-dotnet/issues/3690 😉 

Of course, it can (and should) be addressed in a separate issue. Feel free to resolve this comment.